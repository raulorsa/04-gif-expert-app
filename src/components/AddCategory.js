import React, {useState} from 'react'

import PropTypes from 'prop-types'

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setinputValue(e.target.value);
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length>2){

            //setCategorias( cats => [...cats, inputValue ] );
            setCategorias( cats => [inputValue,...cats ] );

            setinputValue('');
        }
        //console.log('Submit');

        
    }

    return (
        
        <form onSubmit={handleSubmit}>
            {/* <p> {inputValue}</p> */}

            <input 
            type="text"
            value= {inputValue}
            onChange= {handleInputChange}
            
            />
            
        </form>

    )
}

AddCategory.propTypes ={
    setCategorias: PropTypes.func.isRequired
}
