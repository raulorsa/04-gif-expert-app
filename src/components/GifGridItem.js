import React from 'react'

import PropTypes from 'prop-types';


export const GifGridItem = ( {id, title, url} ) => {
    console.log(title);

    return (
        <div className="card">
            <img src={ url} alt={title}/>
            <p> {title}</p>
        </div>
    )
}

GifGridItem.propTypes ={ 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,

}


/*
    1.Enzime
    2.Enzime to Json
    mostrar componentes
    *shallow
    *wrapper
         */
