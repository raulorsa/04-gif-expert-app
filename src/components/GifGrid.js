import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
//import React, {useState, useEffect}  from 'react'
import { GifGridItem } from './GifGridItem';
//import { getGift } from '../helpers/getGifs';

export const GifGrid = ({ category}) => {

    // const [images, setImages] = useState( [])

    const {data, loading} = useFetchGifs( category);

    // useEffect(() =>{
    //     getGift( category).then( imgs => setImages(imgs));
    // },[ category])




    return (
        <>
            <h3  > {category} </h3>

            {loading && <p>Loading</p> }

            <div className="card-grid animate__animated animate__fadeInDown">

                {
                   //console.log(data)
                   // <li key={img.id}> {img.title}</li>
                   
                       data.map( imgs => (

                        <GifGridItem 
                        key={imgs.id}
                        {...imgs}
                        />
                    ))
                }

            </div> 
        </>
    )
}
