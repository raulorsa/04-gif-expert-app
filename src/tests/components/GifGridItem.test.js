
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />',() => {


    const title ="Un Título";
    const url ='http://www.google.es';
    const wrapper = shallow( <GifGridItem title={ title} url={ url}/>);


    test('debe de mostrat el componente correctamente', () => {

        expect( wrapper).toMatchSnapshot();
    })

    test('debe de tener un párrafo con el title',() => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title);
    })
 
 
    test('debe de tener la imagen igual a la url y alt de los props',() => {
        const img = wrapper.find('img');

      //  expect( img.props('src')).toBe(url);
      //  expect( img.props('alt')).toBe(title);
    })

    test('debe de tener amimate__fade in',() => {
        const div = wrapper.find('div');

        const className = div.prop('className');

      //  expect( className.includes('animate__fadeIn')).toBe(true);

    })


})