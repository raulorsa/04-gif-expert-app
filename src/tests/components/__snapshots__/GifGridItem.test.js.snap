// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrat el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Un Título"
    src="http://www.google.es"
  />
  <p>
     
    Un Título
  </p>
</div>
`;
